vxtelegram/telegram.py

```diff
@@ -44,58 +43,81 @@ class TelegramTransport(HttpRpcTransport):
 
     @classmethod
     def agent_factory(cls):
-        """For swapping out the Agent for use in tests"""
+        """
+        For swapping out the Agent for use in tests
+        """
         return Agent(reactor)
 
     @inlineCallbacks
+    def setup_transport(self):
+        yield super(TelegramTransport, self).setup_transport()
+        yield self.add_status_starting()
+
+        config = self.get_static_config()
+        self.api_url = '%s%s' % (config.outbound_url.geturl().rstrip('/'),
+                                 config.bot_token)
+        self.inbound_url = config.inbound_url.geturl()
+        self.bot_username = config.bot_username
+
+        yield self.setup_webhook()
+
+    @inlineCallbacks
     def setup_webhook(self):
         # NOTE: Telegram currently only supports ports 80, 88, 443 and 8443 for
-        #       webhook setup
+        #       webhook setup, and sends requests over HTTPS only
         url = self.get_outbound_url('setWebhook')
         http_client = HTTPClient(self.agent_factory())
+        r = yield http_client.post(
+            url=url,
+            data=json.dumps({'url': self.inbound_url}),
+            headers={'Content-Type': ['application/json']},
+            allow_redirects=False,
+            )
 
-        try:
-            r = yield http_client.post(
-                url=url,
-                data=json.dumps({'url': self.inbound_url}),
-                headers={'Content-Type': ['application/json']}
-                )
         # Telegram redirects our request if our bot token is invalid
-        except ResponseFailed as e:
+        if r.code == http.FOUND:
             self.log.warning(
-                'Webhook setup failed: Invalid token (redirected)\n%s' % e)
+                'Webhook setup failed: Invalid bot token (redirected)')
+            yield self.add_status_bad_webhook(
+                'Invalid bot token (redirected)')
```

Comment:
Considering this case could potentially happen for cases other than invalid tokens, maybe we shouldn't mention the token in the message, and just mention that an unexpected redirect was received.